### PR TITLE
Poc 30

### PR DESCRIPTION
**Pull request by hiqedme**
validate identifiers
check identifier is CCC number
Log viral load error
Log missing UUID error
check if patient data is already synced

### CODE REVIEW

File: packages/eid/app/helpers/getPatientUUID.ts [hunk 1]
```diff
@@ -4,15 +4,32 @@ let CM = config.ConnectionManager.getInstance();
...
-    let result: string = await CM.query(sql, amrsCON);
+    let sql = `select p.uuid from amrs_migration.patient_identifier pi left join amrs_migration.person p on p.person_id=pi.patient_id where pi.identifier='${params}' and pi.voided=0`;
+    let result: any = await CM.query(sql, amrsCON);
+    await CM.releaseConnections(amrsCON);
+    return result;
+  }
+  async getPatientUUIDUsingCCCNo(params: string) {
+    let amrsCON = await CM.getConnectionAmrs();
+    let sql = `select p.uuid from amrs_migration.patient_identifier pi left join amrs_migration.person p on p.person_id=pi.patient_id where pi.identifier='${params}' and pi.identifier_type=28 and pi.voided=0`;
+    let result: any = await CM.query(sql, amrsCON);
```
Comment (drizzentic) [inline packages/eid/app/helpers/getPatientUUID.ts:15]: Can we merge this two functions and make the query string dynamic?

File: packages/eid/app/services/cd4_extractor.ts [hunk 2]
```diff
@@ -38,8 +38,6 @@ export default class ExtractCD4AndPostToETL {
           data.ampath_no
         );
 
-       
-
         if (patientUUID.length > 0) {
           //initialize variables
           let validator = new Validators();
```
Comment (drizzentic) [inline packages/eid/app/services/cd4_extractor.ts:43]: Is this variable being used anywhere?

File: package.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
     "prettier": "^2.1.2"
   },
   "dependencies": {
+    "16": "^0.0.2",
```
Comment (drizzentic) [inline package.json:26]: What is the role of this dependency?

**Review by drizzentic — CHANGES_REQUESTED** [review]
You might want to squash your commits to avoid having lots of them. Or do amends then force push

File: packages/eid/app/helpers/getPatientUUID.ts [hunk 4]
```diff
@@ -4,15 +4,32 @@ let CM = config.ConnectionManager.getInstance();
...
-    let result: string = await CM.query(sql, amrsCON);
+    let sql = `select order_id,encounter_id from amrs_migration.orders where order_number='${params}' and voided=0`;
+    let result: any = await CM.query(sql, amrsCON);
+    await CM.releaseConnections(amrsCON);
+    return result;
+  }
+  async checkPatientDataSync(params: any, uuid: string) {
+   
+    let amrsCON = await CM.getConnectionAmrs();
+    let sql = `select count(encounter_id)  as count from etl.flat_labs_and_imaging where test_datetime='${params.collection_date}' and uuid='${uuid}' and hiv_viral_load = '${params.viral_load}'`;
```
[the remark this answers is not in the document]
Comment (hiqedme) [reply packages/eid/app/helpers/getPatientUUID.ts:29]: confirmed. Uuid = person uuid

File: packages/eid/app/helpers/validators.ts [hunk 5]
```diff
@@ -74,4 +68,48 @@ export default class Validators {
...
+    }
+    // check position of the hyphen
+    else if (identifierNumber.indexOf("-") != 9) {
+      isValid = false;
+    } else {
+      isValid = true;
+    }
+    return isValid;
+  }
+
```
[the remark this answers is not in the document]
Comment (hiqedme) [reply packages/eid/app/helpers/validators.ts:114]: done

File: packages/eid/app/services/cd4_extractor.ts (hunk 2 above)
Comment (hiqedme) [reply packages/eid/app/services/cd4_extractor.ts:43]: removed

File: package.json (hunk 3 above)
Comment (hiqedme) [reply package.json:26]: dependency removed

File: packages/eid/app/helpers/getPatientUUID.ts (hunk 1 above)
Comment (hiqedme) [reply packages/eid/app/helpers/getPatientUUID.ts]: done

File: packages/eid/app/services/cd4_extractor.ts [hunk 6]
```diff
@@ -78,8 +81,8 @@ export default class ExtractCD4AndPostToETL {
         }
       }
       console.log(ResultData);
```
Comment (drizzentic) [inline packages/eid/app/services/cd4_extractor.ts:83]: Remove this console.log

File: packages/eid/app/helpers/getPatientUUID.ts [hunk 7]
```diff
@@ -2,17 +2,29 @@ import config from "@amrs-integrations/core";
...
+    let result: any = await CM.query(sql, amrsCON);
     await CM.releaseConnections(amrsCON);
     return result;
   }
+
   async getPatientOrderNumber(params: string) {
     let amrsCON = await CM.getConnectionAmrs();
-    let sql = `select order_id,encounter_id from amrs.orders where order_number='${params}' and voided=0`;
-    let result: string = await CM.query(sql, amrsCON);
+    let sql = `select order_id,encounter_id from amrs_migration.orders where order_number='${params}' and voided=0`;
```
Comment (drizzentic) [inline packages/eid/app/helpers/getPatientUUID.ts:19]: We can now reference amrs so we can close the pr.

**Review by drizzentic — APPROVED** [review]

File: packages/eid/app/helpers/getPatientUUID.ts (hunk 7 above)
Comment (hiqedme) [reply packages/eid/app/helpers/getPatientUUID.ts:19]: done

File: packages/eid/app/services/cd4_extractor.ts (hunk 6 above)
Comment (hiqedme) [reply packages/eid/app/services/cd4_extractor.ts:83]: done

**Review by drizzentic — APPROVED** [review]

**Pull request merged**